### PR TITLE
Specification: Various integrations from the reference

### PR DESCRIPTION
**Pull request by sjrd**
* Syntax
* Type lambdas and `AnyKind`
* Auto-application
* Union and intersection types

---

This is a rebased and polished version of what we did during a "Spec Day" at the Scala Center. Everything was internally reviewed already.

### CODE REVIEW

File: docs/_docs/internals/syntax.md [hunk 1]
```diff
@@ -75,25 +70,25 @@ floatType        ::=  ‘F’ | ‘f’ | ‘D’ | ‘d’
...
+                   |  alphaid ‘"""’ {[‘"’] [‘"’] char \ (‘"’ | ‘\$’) | escape} {‘"’} ‘"""’
+interpolatedStringPart
                  ::= printableChar \ (‘"’ | ‘$’ | ‘\’) | escape
-escape           ::=  ‘$$’
-                   |  ‘$’ letter { letter | digit }
-                   |  ‘{’ Block  [‘;’ whiteSpace stringFormat whiteSpace] ‘}’
-stringFormat     ::=  {printableChar \ (‘"’ | ‘}’ | ‘ ’ | ‘\t’ | ‘\n’)}
-
-symbolLiteral    ::=  ‘'’ plainid // until 2.13
+escape           ::=  ‘\$\$’
```
Comment (julienrf) [inline docs/_docs/internals/syntax.md:86]: Did you check the rendering?

**Review by julienrf — COMMENTED** [review]
I am not sure about the changes in `docs/_docs/`. I thought we would only edit files in `docs/_spec`.

File: docs/_docs/internals/syntax.md (hunk 1 above)
Comment (sjrd) [reply docs/_docs/internals/syntax.md:86]: The rendering is good:
image

**Comment by sjrd** [comment]
> I am not sure about the changes in `docs/_docs/`. I thought we would only edit files in `docs/_spec`.

@smarter Do you want to comment about the changes in `docs/_docs/`? IIUC they are only aligning with the de facto reality.

**Comment by smarter** [comment]
Yes, they're described in https://github.com/lampepfl/dotty/pull/17383/commits/692b2b376069222b57af8e798d1dff497eb167a7:
> Scala 2 changes compared to the existing Scala 3 spec:
> - Various refactorings
> - Specifying the new Unicode escape handling stuff, this was already implemented
>  in Scala 3 but not part of syntax.md (see https://github.com/lampepfl/dotty/pull/8480).

**Pull request merged**